### ISSUE TITLE
Unable to open file with leading space in file name when passed as a parameter from command line

### ISSUE THREAD

**Issue by martanko** [Microsoft/vscode#29764]
- VSCode Version: 1.13.1
- OS Version: Win10 Pro

Steps to Reproduce:

1. Create a file with leading space, e.g. " my file.txt"
2. Whilst in the text file folder, pass the file name to VS Code from command line, e.g. "c:\Program Files (x86)\Microsoft VS Code\Code.exe" " my file.txt"; VSCode will try to open file "my file.txt" (space removed). 
4. When doing the same with full file path, e.g. "c:\Program Files (x86)\Microsoft VS Code\Code.exe" "c:\ my file.txt"; VSCode will not open any file

Note: When opening from within Code by File->Open, everything works fine

Reproduces without extensions: Yes

**Comment by martanko** [comment]
I have identified the issue in code/node/paths.ts : preparePath() and base/common/paths.ts : isValidBasename().
But there are several more issues with that validation code, e.g. only file name is checked against blacklisted names and characters, but folder names are not, so I'm able to create folders on my (ntfs win) file system that I should not be able to - including trailing spaces, trailing dots and win reserved names.
I have started to fix this, will make a pull request when I find time to finish it; probably will have to split into 2 or 3 separate issues (or find existing ones if already reported) as it's not only "leading space" related + have to test this on a linux mashine as well
